### ISSUE TITLE
Permission issue

### ISSUE THREAD

**Issue by Inspirateur** [Inspirateur/wordy#2]
> For the most of the channels, the bot claims to not have the ability to read past messages, possibly due to [Read Message History] permission missing from the bot. I gave the bot permission to View Channels and to Read Message History, but it still didn't work. Then I added Admin role to the bot, but it still didn't work. 

> However, what is weird that it works on certain channels (there are no special permissions there that cause them to be different). Messages read and from the channels that are read, the bot makes the wordcloud :) But I don't know how to resolve the couldn't read past messages issue. Once again, thank you! :)

![Bildschirmfoto_20241030_123508](https://github.com/user-attachments/assets/ae8f4ef6-becb-4774-9f26-44e5511f1b80)

_Originally posted by @AZMindroma in https://github.com/Inspirateur/wordy/issues/1#issuecomment-2446769171_

**Comment by Inspirateur** [comment]
I'll need to investigate further to understand this issue properly, this is more complicated than just updating the code >.>

**Comment by AZMindroma** [comment]
I understand. By the way, this isn't urgent so you don't need to feel pressured to do it fast. And if I can help investigating in some way like testing different combinations and permissions, you can let me know :)

**Comment by Inspirateur** [comment]
> I understand. By the way, this isn't urgent so you don't need to feel pressured to do it fast. And if I can help investigating in some way like testing different combinations and permissions, you can let me know :)

Yeah, I probably won't be fixing this one today but if you can figure out the difference between channels where it works and channels where it doesn't that would be useful :)

My hunch is that it only works in channel that are public to everyone 🤔 but i didn't check

**Comment by AZMindroma** [comment]
Heya! I've tried finding out what exactly causes the bot to not be able to read, but I haven't found a proper pattern. Some channels share the exact same permissions, yet some are read and some not. It's very weird. I can share my notes below:

# Channel Permission Issue Wordcloud Bot

I have written down all the channels and what permissions these channels have in which the bot is able to scan through the messages. It is to be noted that the bot has admin perms, is on the very top of the role list and has unlimited permissions on the server. 
### Notes:
- Channel X Access Role means a special role is needed to access that channel. 
- Bots Role is a generic role for bots with mere "View Channels" permission.
- If not specified, the rest of the permissionset is set to NEUTRAL /
- At the end of the permissions I have also added the message count of the channel.
- The categories don't have any permission overrides.
- The names are just for organization purposes. The first three letters show what category the channel is in and the last three letters is a codename for the channel 
## Channels that can be read by the bot
### Channel COA-MEM: 
**everyone**: Send Messages OFF, Add Reactions OFF
**Messages**: 41
### Channel ACT-SEA:
**everyone**: View Channel OFF
**Channel ACT-SEA & ACT-FRM Access Role**: View Channel ON 
**Messages**: 50232
### Channel ARC-OWS:
**everyone**: View Channel OFF
**Messages**: 398
### Channel GAL-SHO: 
**everyone**: All NEUTRAL /
**Messages**: 202
### Channel ACT-ZWS:
**everyone**: ALL NEUTRAL /
**Messages**: 86
### Channel UGR-TFC:
**everyone**: View Channel OFF
**Bots role**: View Channel ON
**Channel UGR-TFC Access Role**: View Channel ON
**Messages**: 6680
### Channel STF-CLO:
**everyone**: View Channel OFF
**Bots role**: View Channel ON
**Messages**: 11848
### Channel STF-SSD:
**everyone**: View Channel OFF
**Bots role**: View Channel ON
**Messages**: 11
### Channel ACT-OCS:
**everyone**: All NEUTRAL /
**Messages**: 80
## Channels that cannot be read by the bot
### Channel COA-INF:
**everyone**: View Channel ON
**Messages**: 6
### Channel COA-ANN:
**everyone**: Send Messages OFF
**Messages**: 22
### Channel COA-COM:
**everyone**: Send Messages ON, Create Public Threads OFF
**Messages**: 246
### Channel COA-ITR:
**everyone**: All NEUTRAL /
**Messages**: 30
### Channel GAR-QUO
**everyone**: Send Messages OFF
**Bots role**: Send Messages ON
**Messages**: 229
### Channel GAR-POF
**everyone**: Send Messages OFF
**Bots role**: Send Messages ON
**Messages**: 48
### Channel GAR-POL
**everyone**: All NEUTRAL /
**Messages**: 83
### Forum GAR-HUT
**everyone**: All NEUTRAL /
**Messages**: 584
### Channel COH-FIR
**everyone**: All NEUTRAL /
**Messages**: 249369
### Channel COH-SRS
**everyone**: View Channels OFF
**Channel COH-SRS Access Role**: View Channel ON
**Messages**: 2880
### Channel ACT-STR
**everyone**: View Channels OFF
**Channel ACT-STR Access Role**: View Channels ON
**Messages**: 115
### Channel ACT-FRM
**everyone**: View Channel OFF
**Channel ACT-SEA & ACT-FRM Access Role**: View Channel ON 
**Messages**: 83
### Channel STF-STF
**everyone**: View Channel OFF
**Messages**: 172

**Comment by Inspirateur** [comment]
This is very weird ... Can you try kicking the bot out and re-inviting it ?

**Comment by Raindove** [comment]
Hello! Was there ever a fix for this? I'm running into the same issue. Thanks!

**Comment by Inspirateur** [comment]
> Hello! Was there ever a fix for this? I'm running into the same issue. Thanks!

Not yet :v
I'll investigate further once I get back from vacations

**Comment by Kalvin126** [comment]
Im hitting this issue as well. Is there some way to have it re-try? Or does it re-try on some cadence? Or does it re-try on restarts?

**Comment by Kalvin126** [comment]
If it helps, I don't see a "Read a new message" trace logs for message for some channels too.

**Comment by Inspirateur** [comment]
yeah i'll look into it when i get more bandwidth, i'm very busy irl right now

**Comment by Inspirateur** [comment]
Ok so turns out this is just a permission issue with private channels, I improved the warning messages but basically: if you're using private channels make sure the bot has explicit access to it ! Its role won't be enough.

I'm also adding it to the readme to help people having the same issue :)
